src/main/java/org/ibp/api/java/impl/middleware/name/validator/NameTypeValidator.java

```diff
@@ -0,0 +1,81 @@
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@Component
+public class NameTypeValidator {
+
+	private static final Integer CODE_OR_NAME_MAX_LENGTH = 50;
+	private static final Integer DESCRPTION_MAX_LENGTH = 255;
+	private BindingResult errors;
+
+	@Autowired
+	GermplasmNameTypeService germplasmNameTypeService;
+
+	public void validate(final GermplasmNameTypeRequestDTO germplasmNameTypeRequestDTO) {
+		this.errors = new MapBindingResult(new HashMap<>(), GermplasmNameTypeRequestDTO.class.getName());
+
+		if (StringUtils.isBlank(germplasmNameTypeRequestDTO.getCode())) {
+			this.errors.reject("name.type.field.invalid", new String[] {"code"}, "");
+			throw new ApiRequestValidationException(this.errors.getAllErrors());
+
+		}
+
+		if (germplasmNameTypeRequestDTO.getCode().length() > NameTypeValidator.CODE_OR_NAME_MAX_LENGTH) {
+			this.errors.reject("name.type.field.length.invalid", new String[] {"code", NameTypeValidator.CODE_OR_NAME_MAX_LENGTH.toString()}, "");
+			throw new ApiRequestValidationException(this.errors.getAllErrors());
+		}
+
+		this.validateIfCodeAlreadyExists(germplasmNameTypeRequestDTO.getCode());
+
+		if (StringUtils.isBlank(germplasmNameTypeRequestDTO.getName())) {
+			this.errors.reject("name.type.field.invalid", new String[] {"name"}, "");
+			throw new ApiRequestValidationException(this.errors.getAllErrors());
+		}
+
+		if (germplasmNameTypeRequestDTO.getName().length() > NameTypeValidator.CODE_OR_NAME_MAX_LENGTH) {
+			this.errors.reject("name.type.field.length.invalid", new String[] {"name", NameTypeValidator.CODE_OR_NAME_MAX_LENGTH.toString()}, "");
+			throw new ApiRequestValidationException(this.errors.getAllErrors());
+		}
+
+		this.validateIfNameAlreadyExists(germplasmNameTypeRequestDTO.getName());
+
+		if (StringUtils.isNotBlank(germplasmNameTypeRequestDTO.getDescription()) && //
+			germplasmNameTypeRequestDTO.getDescription().length() > NameTypeValidator.DESCRPTION_MAX_LENGTH) {
+			this.errors.reject("name.type.field.length.invalid",
+				new String[] {"description", NameTypeValidator.DESCRPTION_MAX_LENGTH.toString()}, "");
+			throw new ApiRequestValidationException(this.errors.getAllErrors());
+		}
+	}
+
+	protected void validateIfCodeAlreadyExists(final String code) {
+		final Set<String> codes = new HashSet<>(Arrays.asList(code));
+		final List<GermplasmNameTypeDTO> nameTypeDTOS = this.germplasmNameTypeService.filterGermplasmNameTypes(codes);
+		if (nameTypeDTOS != null && !nameTypeDTOS.isEmpty()) {
+			this.errors.reject("name.type.code.invalid", new String[] {code}, "");
+			throw new ApiRequestValidationException(this.errors.getAllErrors());
+		}
+	}
+
+	protected void validateIfNameAlreadyExists(final String name) {
+		final List<GermplasmNameTypeDTO> nameTypeDTOS = this.germplasmNameTypeService.filterGermplasmNameTypesByName(name);
+		if (nameTypeDTOS != null && !nameTypeDTOS.isEmpty()) {
```

Comment:
seems like nameTypeDTOS != null is not needed. The filter function is never returning null